decompy/DataGathering/CreateLocalData.py

```diff
@@ -1,87 +1,492 @@
+        :param start_date: date to start or pick back up formatted "%Y-%m-%d
+        :type: str
+        :param end_date: date to end on formatted "%Y-%m-%d
+        :type: str
+        :param start_page: page to start or pick up from.
+        :type: int
+        :param end_page: page to end or pick up from.
+        :type: int
+        :return: void
+        """
+
+        # format date time for start and end date.
+        if start_date is None:
+            start_date = "2013-04-11"
+
+        if end_date is None:
+            self.repo_end_date = datetime.today()
+        else:
+            self.repo_end_date = datetime.strptime(end_date, "%Y-%m-%d").date()
+
+        # start date end date
+        self.repo_start_date = datetime.strptime(start_date, "%Y-%m-%d").date()
+
+        # add 1 day
+        while self.repo_start_date <= self.repo_end_date:
+            self.repo_start_date = self.repo_start_date + timedelta(days=1)
+
+            try:
+                # only do 100 repos at a time (1 page) for safety on the json file
+                self.stage1_gather_repo_meta(self.repo_start_date, start_page, end_page)
+
+                # skip this, no info
+                if self.skip is False:
+                    self.stage2_get_repos(False, self.username, self.password)
+                    self.stage3_filter_files()
+                    self.stage4_generate_llvm()
+                    self.stage5_insert_database()
+
+                    files = os.listdir(self.folder)
+                    for f in files:
+                        try:
+                            shutil.move(os.path.join(self.folder, f), os.path.join(self.dest_folder, f))
+                        except Exception as e:
+                            print("moving files error:", e)
+                            pass
+                    # remove filtered files in repositories
+                    for f in files:
+                        try:
+                            if os.path.exists(f):
+                                shutil.rmtree(f)
+                        except Exception as e:
+                            print("removing files error", e)
+                            pass
+
+                # reset skip
+                self.skip = False
+
+            except Exception as e:
+                print("Running all stages error:", e)
+                pass
+
 
 if __name__ == "__main__":
+    # cld = CreateLocalData()
+    # cld.stage1_gather_repo_meta()
+    # cld.stage2_get_repos()
+    # cld.stage3_filter_files()
+    # cld.stage4_generate_llvm()
+    # cld.stage5_insert_database()
     cld = CreateLocalData()
-    cld.stage1()
-    print("stage 1 done")
-    cld.stage2()
-    print("stage 2 done")
-    cld.stage3()
-    print("stage 3 done")
-    cld.stage4()
-    print("stage 4 done")
+    print("Beginning to download all C data in 10 seconds...")
+    time.sleep(10)
+    cld.all_stages_increment("2014-04-11", "2019-04-11", 1, 10)
```

Comment:
This file is terrible OOP. It's one monolithic procedure really. It's ~500 lines

Reply:
yeah it's bad for now, but only served 1 real purpose and can be split into itself... Maybe another user story in the future for polishing can fix this.